src/main.js

```diff
@@ -7,155 +7,155 @@ Only expose a tiny API to keep internal routing safe
 
 @singleton Aviator
 **/
-var Aviator = {
-
-  /**
-  @property pushStateEnabled
-  @type {Boolean}
-  @default true if the browser supports pushState
-  **/
-  pushStateEnabled: ( typeof window !== 'undefined' && typeof window.history !== 'undefined' ) ? 'pushState' in window.history : false,
-
-  /**
-  @property linkSelector
-  @type {String}
-  @default 'a.navigate'
-  **/
-  linkSelector: 'a.navigate',
-
-  /**
-  the root of the uri from which routing will append to
-
-  @property root
-  @type {String}
-  @default ''
-  **/
-  root: '',
-
-  /**
-  @property _navigator
-  @type {Navigator}
-
-  @private
-  **/
-  _navigator: new Navigator(),
-
-  /**
-  @method setRoutes
-  @param {Object} routes
-  **/
-  setRoutes: function (routes) {
-    this._navigator.setRoutes(routes);
-  },
-
-  /**
-  dispatches routes to targets and sets up event handlers
-
-  @method dispatch
-  **/
-  dispatch: function (config) {
-    var navigator = this._navigator, navigatorConfig;
-
-    config = config || {};
-
-    navigatorConfig = {
-      pushStateEnabled: this.pushStateEnabled,
-      linkSelector:     this.linkSelector,
-      root:             this.root
-    };
-
-    // for now, just expose config limited config that we need
-    if (typeof config.inBrowser !== 'undefined' && !config.inBrowser) {
-    	navigatorConfig._inBrowser = false;
-    }
-    if (typeof config.modifyUrl !== 'undefined' && !config.modifyUrl) {
-    	navigatorConfig._modifyUrl = false;
-    }
-    if (typeof config.silent !== 'undefined' && config.silent) {
-      navigatorConfig._silent = true;
-    }
+function AviatorConstructor() { }
```

Comment:
I would keep the `Constructor` word out of the variable name and just keep it as `Aviator` it's implied that this is the constructor. Also could we move the properties of this "class" (e.g. `pushStateEnabled`, `linkSelector`, `_navigator`, etc.) here instead of accessing the `.prototype` directly? I would declare the properties inside here and keep the functions the way you have it, assigning it using the `.prototype`